packages/block-editor/src/components/grid/use-grid-layout-sync.js

```diff
@@ -11,31 +11,22 @@ import { store as blockEditorStore } from '../../store';
 import { GridRect } from './utils';
 
 export function useGridLayoutSync( { clientId: gridClientId } ) {
-	const { gridLayout, blockOrder } = useSelect(
-		( select ) => {
-			const { getBlockAttributes, getBlockOrder } =
-				select( blockEditorStore );
-			return {
-				gridLayout: getBlockAttributes( gridClientId ).layout ?? {},
-				blockOrder: getBlockOrder( gridClientId ),
-			};
-		},
-		[ gridClientId ]
-	);
-
-	const { getBlockAttributes } = useSelect( blockEditorStore );
+	const { getBlockOrder, getBlockAttributes } = useSelect( blockEditorStore );
 	const { updateBlockAttributes, __unstableMarkNextChangeAsNotPersistent } =
 		useDispatch( blockEditorStore );
 
+	const blockOrder = getBlockOrder( gridClientId );
+
 	useEffect( () => {
+		const gridLayout = getBlockAttributes( gridClientId ).layout ?? {};
 		const updates = {};
-
-		const { columnCount, rowCount = 2 } = gridLayout;
-		const isManualGrid = !! columnCount;
+		const { columnCount, rowCount, manualPlacement } = gridLayout;
+		const isManualGrid = !! manualPlacement;
 
 		if ( isManualGrid ) {
 			const rects = [];
 			let cellsTaken = 0;
+			let lowestRowEnd = 0;
```

Comment:
This is a somewhat confusing name because "lowest row end" could mean the bottom-most row or the smallest "row end".

Reply:
`bottomestRow`? 😄